### ISSUE TITLE
Adding offline map tile cache as layer

### ISSUE THREAD

**Issue by MINDoSOFT** [sothawo/mapjfx#26]
Hello I am currently using [JMapViewer](https://github.com/balloob/JMapViewer) which has the ability to use an offline tile source as a layer.

The offline tile source has the following structure:
/zoomLevel/X/Y.png

> [Source](https://openlayers.org/workshop/en/layers/cached.html)
> ol.source.XYZ
> 
> The Web Map Service specification allows a lot of flexibility in terms of what a client can request. Without constraints, this makes caching difficult or impossible in practice.
> 
> At the opposite extreme, a service might offer tiles only at a fixed set of zoom levels and only for a regular grid. These can be generalized as tiled layers with an XYZ source - you can consider X and Y to indicate the column and row of the grid and Z to represent the zoom level.
> 

Is it possible to add the functionality of these offline tile sources for mapjfx? 

Thanks in advance.

**Comment by sothawo** [comment]
do you have an example of how to integrate the offline tile data into OpenLayers? Or do you have some sample data? When this is working with OpenLayers, the integration into mapjfx is easy. If you already have information about that it would be helpful, because this would save me quite some time in research.

**Comment by MINDoSOFT** [comment]
Basically what we wanted was to have access to a map without access to the internet.

The method that I describe has the drawback that you must generate the tiles beforehand which takes quite some time even for small countries like mine (Cyprus). I generated my tiles by downloading osm.pbf files from [here](http://download.geofabrik.de/) and then rendering the tiles using [Maperitive](http://maperitive.net/).

So I went with the solution to use the [mapsforge](https://github.com/mapsforge/mapsforge) library which can use a file that contains the OSM vector data of the map which is rather small. Even for Germany it is 1.8 GB (and germany is a huge country). The only thing with mapsforge is that it supports awt for now. They have an [issue open to make it JavaFX](https://github.com/mapsforge/mapsforge/issues/928).
